### PR TITLE
Turret testing

### PR DESCRIPTION
**Pull request by lydiaxing**
not ready to merge -- need to edit talon methods

### CODE REVIEW

File: src/com/nutrons/framework/inputs/Serial.java [hunk 1]
```diff
@@ -65,13 +72,15 @@ public Serial(int baudrate,
...
 
-    this.dataStream = toFlow(() -> {
+    Supplier<byte[]> suppler = (Supplier<byte[]>)() -> {
       if (serial.getBytesReceived() > this.bufferSize) { //Clear out old values
         serial.reset();
       }
       return serial.read(packetLength);
-    }).filter(x -> x.length == packetLength);
+    };
+    this.dataStream = toFlow(new IntervalCache<byte[]>(100, suppler)).filter(x -> x.length == packetLength);
```
Comment (MiloDavis) [inline src/com/nutrons/framework/inputs/Serial.java]: Newline before `filter`

**Review by MiloDavis — CHANGES_REQUESTED** [review]

**Comment by lydiaxing** [comment]
Fixed style changes

File: src/com/nutrons/framework/subsystems/WpiSmartDashboard.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.nutrons.framework.subsystems;
```
Comment (MiloDavis) [inline src/com/nutrons/framework/subsystems/WpiSmartDashboard.java:1]: Add a TODO comment for us to implement the remaining functionality.  There's a bunch more stuff in the Java class and we might want to have it during competitions.

File: src/com/nutrons/framework/util/IntervalCache.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Override
+    public T get() {
+        if(needUpdate()){
+            return update();
+        }else{
+            return lastValue;
+        }
+    }
+}
```
Comment (MiloDavis) [inline src/com/nutrons/framework/util/IntervalCache.java]: Add newline at end of file

**Review by MiloDavis — CHANGES_REQUESTED** [review]

File: src/com/nutrons/framework/controllers/Talon.java [hunk 4]
```diff
@@ -92,4 +97,17 @@ public void accept(ControllerEvent event) {
   int id() {
     return this.talon.getDeviceID();
   }
+
+  void setOutputVoltage(double min, double max) {
+    this.talon.configNominalOutputVoltage(Math.max(min, 0.0), Math.min(max, 0.0));
```
Comment (MiloDavis) [inline src/com/nutrons/framework/controllers/Talon.java]: bug here

**Review by MiloDavis — DISMISSED** [review]

**Review by kreeve — APPROVED** [review]

**Pull request merged**